components/schema-migrator/migrations/201908141300_initialize_schema.up.sql

```diff
@@ -0,0 +1,247 @@
+    app_id uuid NOT NULL,
+    foreign key (tenant_id, app_id) references applications (tenant_id, id) ON DELETE CASCADE,
+    title varchar(256) NOT NULL,
+    display_name varchar(256) NOT NULL,
+    description text NOT NULL,
+    format document_format NOT NULL,
+    kind varchar(256),
+    data text
+);
+
+CREATE INDEX ON documents (tenant_id);
+CREATE UNIQUE INDEX ON documents (tenant_id, id);
+
+-- Label Definition
+
+CREATE TABLE label_definitions (
+    id uuid PRIMARY KEY,
+    tenant_id uuid NOT NULL,
+    key varchar(256) NOT NULL,
+    schema jsonb
+);
+
+CREATE INDEX ON label_definitions (tenant_id);
+CREATE UNIQUE INDEX ON label_definitions (tenant_id, key);
+
+-- Label
+
+CREATE TABLE labels (
+    id uuid PRIMARY KEY,
+    tenant_id uuid NOT NULL,
+    app_id uuid, -- TODO: Update when Applications switch to DB repository
+    runtime_id uuid,
+    foreign key (tenant_id, runtime_id) references runtimes (tenant_id, id) ON DELETE CASCADE,
+    key varchar(256) NOT NULL,
+    value jsonb,
+    CONSTRAINT valid_refs CHECK (app_id IS NOT NULL OR runtime_id IS NOT NULL)
+);
+
+CREATE INDEX ON labels (tenant_id);
+CREATE UNIQUE INDEX ON labels (tenant_id, key, runtime_id, app_id);
+CREATE UNIQUE INDEX ON labels (tenant_id, id);
+
+-- Fetch Request
+
+CREATE TYPE fetch_request_status_condition AS ENUM (
+    'INITIAL',
+    'SUCCEEDED',
+    'FAILED'
+);
+
+CREATE TYPE fetch_request_mode AS ENUM (
+    'SINGLE',
+    'PACKAGE',
+    'INDEX'
+);
+
+CREATE TABLE fetch_requests (
+    id uuid PRIMARY KEY,
+    tenant_id uuid NOT NULL,
+
+    api_def_id uuid,
+    foreign key (tenant_id, api_def_id) references api_definitions (tenant_id, id) ON DELETE CASCADE,
+    event_api_def_id uuid,
+    foreign key (tenant_id, event_api_def_id) references event_api_definitions (tenant_id, id) ON DELETE CASCADE,
+    document_id uuid,
+    foreign key (tenant_id, document_id) references documents (tenant_id, id) ON DELETE CASCADE,
+
+    url varchar(256) NOT NULL,
+    auth jsonb,
+    mode fetch_request_mode NOT NULL,
+    filter varchar(256),
+    status_condition fetch_request_status_condition DEFAULT 'INITIAL' ::fetch_request_status_condition NOT NULL,
+    status_timestamp timestamp NOT NULL,
+    CONSTRAINT valid_refs CHECK (api_def_id IS NOT NULL OR event_api_def_id IS NOT NULL OR document_id IS NOT NULL)
+);
+
+CREATE INDEX ON fetch_requests (tenant_id);
+CREATE UNIQUE INDEX ON fetch_requests (tenant_id, api_def_id, event_api_def_id, document_id);
+CREATE UNIQUE INDEX ON fetch_requests (tenant_id, id);
+
+ALTER TABLE api_definitions
+    ADD spec_fetch_request_id uuid,
+    ADD foreign key (tenant_id, spec_fetch_request_id) references fetch_requests (tenant_id, id) ON DELETE CASCADE;
+ALTER TABLE event_api_definitions
+    ADD spec_fetch_request_id uuid,
+    ADD foreign key (tenant_id, spec_fetch_request_id) references fetch_requests (tenant_id, id) ON DELETE CASCADE;
+ALTER TABLE documents
+    ADD fetch_request_id uuid,
```

Comment:
In this case reference to `fetch_request` `id` differs from others, like in `api_definitions`?
Because, in `documents` we have name `fetch_request_id` and in `api_definition` we have `spec_fetch_request_id`.

Reply:
yes, because in API and Event API we have specs